pkg/okteto/kubetoken/cache.go

```diff
@@ -0,0 +1,122 @@
+package kubetoken
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/okteto/okteto/pkg/log"
+	authenticationv1 "k8s.io/api/authentication/v1"
+)
+
+const cacheIsCorrupted = "cache is corrupted"
+
+type stringStore interface {
+	Get() ([]byte, error)
+	Set([]byte) error
+}
+
+type Cache struct {
+	StringStore stringStore
+	Now         func() time.Time
+	Debug       func(string, ...interface{})
+}
+
+func NewCache(fileName string) *Cache {
+	return &Cache{
+		StringStore: NewFileByteStore(fileName),
+		Now:         time.Now,
+		Debug:       log.Debugf,
+	}
+}
+
+type storeRegister struct {
+	ContextName string                        `json:"context"`
+	Namespace   string                        `json:"namespace"`
+	Token       authenticationv1.TokenRequest `json:"token"`
+}
+
+func (c *Cache) read() ([]storeRegister, error) {
+	contents, err := c.StringStore.Get()
+	if err != nil {
+		return nil, err
```

Comment:
Let's add some information about the errors. Otherwise it's missing information when we try to debug them:
```suggestion
		return nil, fmt.Errorf("error while trying to get kubetoken cache: %w", err)
```